exp/internal/controllers/machinepool_controller_noderef.go

```diff
@@ -110,11 +110,20 @@ func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *
 			clusterv1.OwnerKindAnnotation:        mp.Kind,
 			clusterv1.OwnerNameAnnotation:        mp.Name,
 		}
-		if annotations.AddAnnotations(node, desired) {
-			if err := patchHelper.Patch(ctx, node); err != nil {
-				log.V(2).Info("Failed patch node to set annotations", "err", err, "node name", node.Name)
-				return ctrl.Result{}, err
+		// Add annotations.
+		annotations.AddAnnotations(node, desired)
```

Comment:
Just for confirmation, even if we are always calling patch helper, a patch is going to be sent to the api server only if there is a diff, is this right?
worndering if we can make this explict by doing somenting like in the previous implementation

```
if annotations.AddAnnotations(node, desired) || taints.AddTaints(node, desired) {
    patch...
}
```

Reply:
> Just for confirmation, even if we are always calling patch helper, a patch is going to be sent to the api server only if there is a diff, is this right?

Yes!

patchHelper checks if there is a diff before calling Patch on the CR client

Reply:
I don't have strong opinions on this.  I am fine with the current implementation as the patch helper takes care of it already.